Client/mods/deathmatch/logic/CClientGame.cpp

```diff
@@ -6402,10 +6400,30 @@ void CClientGame::StaticWorldSoundHandler ( uint uiGroup, uint uiIndex )
 // Handle callback from CAudioSA when a world sound is played
 //
 //////////////////////////////////////////////////////////////////
-void CClientGame::WorldSoundHandler ( uint uiGroup, uint uiIndex )
+bool CClientGame::WorldSoundHandler ( uint uiGroup, uint uiIndex, CEntitySAInterface* pEntityInterface, const CVector& vecPosition )
 {
     if ( m_bShowSound )
         m_pScriptDebugging->LogInformation ( NULL, "%s - World sound group:%d index:%d", *GetLocalTimeString ( false, true ), uiGroup, uiIndex );
+
+    if ( pEntityInterface ) {
+        CClientEntity* pEntity = g_pClientGame->GetGameEntityXRefManager()->FindClientEntity( pEntityInterface );
+
+        if ( pEntity && ( pEntity->GetType() == CCLIENTPED || pEntity->GetType() == CCLIENTPLAYER ) ) {
```

Comment:
- The pull request description tackles this issue: The hooked GTA:SA function doesn't handle a lot of sounds and that's why it is restricted to player/ped entities, because you can only cancel these.
- The source element can never be **nil**, it's either the source element or root